src/internal_events/wasm/compilation.rs

```diff
@@ -74,7 +74,7 @@ impl InternalEvent for WasmCompilationProgress {
     }
 
     fn emit_metrics(&self) {
-        counter!("wasm_compilation", 1,
+        counter!("wasm_compilation_total", 1,
```

Comment:
Not directly related to this PR (though it could impact the naming), but it is unclear to me exactly what this metric indicates. Does it indicate the count of state transitions?

cc/ @Hoverbear